### DISCUSSION TITLE
Version 2.0 wishlist

### DISCUSSION THREAD

**Discussion by selwin** [rq/rq#1807] [General]
- [ ] Deprecate `exc_handler` and `exception_handlers` (possibly) args on `Worker.__init__` 
- [ ] Requires Redis server >= 5.0 for (Redis streams)
- [ ] Deprecate `prepare_for_work` from `Worker.__init__`
- [ ] Deprecate `log_job_description` args on `Worker.__init__`
- [ ] Replace `RoundRobinWorker` and `RandomWorker` with `DequeueStrategy`
- [ ] Rename `default_worker_ttl ` argument to `Worker` to `dequeue_ttl`

**Comment by ccrvlh** [comment]
@selwin can we move this to a discussion? this is the sort of issue that will never get closed :)

**Comment by selwin** [comment]
Hey, I'm ok with continuing this on the discussion, but I'd like to have a checklist for a list of planned deprecations so I don't forget.

For most of these things, we'll need to have a replacement ready way in advance (like `DequeueStrategy`) so people have time to move into them.
